keps/sig-storage/3762-persistent-volume-last-phase-transition-time/README.md

```diff
@@ -0,0 +1,879 @@
+- [ ] (R) Production readiness review completed
+- [ ] (R) Production readiness review approved
+- [ ] "Implementation History" section is up-to-date for milestone
+- [ ] User-facing documentation has been created in [kubernetes/website], for publication to [kubernetes.io]
+- [ ] Supporting documentation—e.g., additional design documents, links to mailing list discussions/SIG meetings, relevant PRs/issues, release notes
+
+<!--
+**Note:** This checklist is iterative and should be reviewed and updated every time this enhancement is being considered for a milestone.
+-->
+
+[kubernetes.io]: https://kubernetes.io/
+[kubernetes/enhancements]: https://git.k8s.io/enhancements
+[kubernetes/kubernetes]: https://git.k8s.io/kubernetes
+[kubernetes/website]: https://git.k8s.io/website
+
+## Summary
+
+<!--
+This section is incredibly important for producing high-quality, user-focused
+documentation such as release notes or a development roadmap. It should be
+possible to collect this information before implementation begins, in order to
+avoid requiring implementors to split their attention between writing release
+notes and implementing the feature itself. KEP editors and SIG Docs
+should help to ensure that the tone and content of the `Summary` section is
+useful for a wide audience.
+
+A good summary is probably at least a paragraph in length.
+
+Both in this section and below, follow the guidelines of the [documentation
+style guide]. In particular, wrap lines to a reasonable length, to make it
+easier for reviewers to cite specific portions, and to minimize diff churn on
+updates.
+
+[documentation style guide]: https://github.com/kubernetes/community/blob/master/contributors/guide/style-guide.md
+-->
+
+We want to add a new PersistentVolumeStatus field, which would hold a timestamp of when a PersistentVolume last
+transitioned to a different phase.
+
+## Motivation
+
+Some users have experienced data loss when using `Delete` retain policy and reverted to a safer `Retain` policy.
+With `Retain` policy all volumes that are retained and left unclaimed have their phase is set to `Released`.
+As the released volumes pile up over time admins want to perform manual cleanup based on the time when the volume was
+last used, which is when the volume transitioned to `Released` phase.
+
+We can approach the solution in a more generic way and record a timestamp of when the volume transitioned to any phase,
+not just to `Released` phase. This allows anyone, incl. our perf tests, to measure time e.g. between a PV `Pending` and `Bound`
+
+<!--
+This section is for explicitly listing the motivation, goals, and non-goals of
+this KEP.  Describe why the change is important and the benefits to users. The
+motivation section can optionally provide links to [experience reports] to
+demonstrate the interest in a KEP within the wider Kubernetes community.
+
+[experience reports]: https://github.com/golang/go/wiki/ExperienceReports
+-->
+
+### Goals
```

Comment:
Since one of the use cases for this can be for general observability and defining SLOs, should we also consider other volume operations like Attach/Detach?

Reply:
Being able to observe phase change timestamps is directly related to Attach/Detach operations. In other words Attach/Detach will be more observable if admins can see when the volume transitioned to `Bound`/`Released` phase. I think it's ok to add this as one of the goals, did I understand your proposal correctly?